azureDatabaseUtils/src/main/java/bio/terra/workspace/azureDatabaseUtils/storage/AzureBlobStorage.java

```diff
@@ -0,0 +1,106 @@
+
+  backups will be stored inside the workspace container in a path determined by function GenerateBackupFilename
+  which exists within BackupService.java.
+   */
+  @Override
+  public void streamOutputToBlobStorage(
+      InputStream fromStream, String blobName, String workspaceId, String blobstorageDetails) {
+    BlobContainerClient blobContainerClient =
+        constructBlockBlobClient(workspaceId, blobstorageDetails);
+    // https://learn.microsoft.com/en-us/java/api/overview/azure/storage-blob-readme?view=azure-java-stable#upload-a-blob-via-an-outputstream
+    try (BufferedOutputStream blobOS =
+        new BufferedOutputStream(
+            blobContainerClient
+                .getBlobClient(blobName)
+                .getBlockBlobClient()
+                .getBlobOutputStream())) {
+      try (BufferedReader bufferedReader =
+          new BufferedReader(new InputStreamReader(fromStream, StandardCharsets.UTF_8))) {
+        int line;
+        while ((line = bufferedReader.read()) != -1) {
+          blobOS.write(line);
+        }
+      }
+    } catch (IOException ioEx) {
+      throw new LaunchProcessException("Error streaming output of child process", ioEx);
+    }
+  }
+
+  @Override
+  public void streamInputFromBlobStorage(
+      OutputStream toStream, String blobName, String workspaceId, String blobstorageDetails) {
+    BlobContainerClient blobContainerClient =
+        constructBlockBlobClient(workspaceId, blobstorageDetails);
+    try (toStream) {
+      blobContainerClient.getBlobClient(blobName).downloadStream(toStream);
+    } catch (IOException ioEx) {
+      throw new LaunchProcessException("Error streaming input to child process", ioEx);
+    }
+  }
+
+  private BlobContainerClient constructBlockBlobClient(
+      String workspaceId, String blobstorageDetails) {
+    // TODO: determine where blobstorageDetails should come from.
+    // This implementation is 95% copied from WDS's, in which they call WSM's APIs to retrieve a url
+    // with an included SAS token.
+    // The implementation here assumes that this url+token will be passed in by WSM itself when it
+    // ultimately calls the pgDump function.
+
+    BlobServiceClient blobServiceClient =
+        new BlobServiceClientBuilder().endpoint(blobstorageDetails).buildClient();
+
+    try {
+      // the way storage containers are set up in a workspace are as follows:
+      // billing project gets a single azure storage account
+      // each workspace gets a container inside of that storage account to keep its data
+      return blobServiceClient.getBlobContainerClient("sc-" + workspaceId);
+    } catch (BlobStorageException e) {
+      // if the default workspace container doesn't exist, something went horribly wrong
+      logger.error("Default storage container missing for workspace id {}", workspaceId);
+      throw (e);
+    }
+  }
+
+  @Override
+  public void deleteBlob(String blobFile, String workspaceId, String authToken) {
```

Comment:
is this used?

Reply:
good call... not yet, but I think it will be. I'll remove for now and reintroduce it when needed